### PR TITLE
Windows bootstrap support

### LINKED ISSUE #25224: Fix mach bootstrap on Windows
According to jdm on the IRC, mach bootstrap doesn't do anything useful, because every build-related command is already triggering [windows_msvc](https://github.com/servo/servo/blob/master/python/servo/bootstrap.py#L275).

Invoking bootstrap however, is done by https://github.com/servo/servo/blob/master/python/servo/bootstrap.py#L396 which in turn is called from https://github.com/servo/servo/blob/master/python/mach_bootstrap.py#L241

The following exception happens when ./mach bootstrap is invoked:
```
$ ./mach bootstrap
Traceback (most recent call last):
  File "./mach", line 103, in <module>
    main(sys.argv)
  File "./mach", line 31, in main
    sys.exit(mach_bootstrap.bootstrap_command_only(topdir))
  File "e:\Marc\Documents\Coding\servo\python\mach_bootstrap.py", line 245, in bootstrap_command_only
    bootstrap(context, force)
  File "e:\Marc\Documents\Coding\servo\python\servo\bootstrap.py", line 413, in bootstrap
    return bootstrapper(context, force=force)
  File "e:\Marc\Documents\Coding\servo\python\servo\bootstrap.py", line 278, in windows_msvc
    deps_dir = os.path.join(context.sharedir, "msvc-dependencies")
AttributeError: 'DummyContext' object has no attribute 'sharedir'
```
I think the solution to this is adding `context.sharedir = os.path.join(context.topdir, ".servo")` to the lastly linked code in mach_bootstrap.py, because other build tasks indeed _do_ download the dependencies to `.servo` relative to the topdir.

Note that this would affect windows and linux alike, but it seems this sharedir is only requested from salt and windows and not linux, which installs the packets globally using apt.

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #25224
- [x] These changes do not require tests because they are just support script changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,9 +43,12 @@ manually, try the [manual build setup][manual-build].
 
  - Download and run [`rustup-init.exe`](https://win.rustup.rs/) then follow the onscreen instructions.
  - Install [chocolatey](https://chocolatey.org/)
- - Run `choco install support\windows\chocolatey.config`
-   *This will install CMake, Git, LLVM, Ninja, NuGet, Python and the Visual Studio 2019 Build Tools.*
- - Run `mach bootstrap-gstreamer`
+ - Run `mach bootstrap`
+  - *This will install CMake, Git, LLVM, Ninja, NuGet, Python and the Visual Studio 2019 Build Tools
```
Comment (mukilan) [inline README.md:47]: Will LLVM be installed via choco as this comment suggests? I see we have removed it in choco.config

**Review by mukilan — APPROVED** [review]
Nice!

File: README.md (hunk 1 above)
Comment (mrobinson) [reply README.md]: Oh, nice catch. I've fixed this now.

**Comment by mrobinson** [comment]
@bors-servo retry
 - Looks like #23168 or #27315

**Pull request merged**